### ISSUE TITLE
Incorrect total number of results

### ISSUE THREAD

**Issue by corrupt952** [instrumenta/conftest#275]
When testing with this tool, i saw incorrect results.

* conftest ... v0.18.0(v0.17.1 and v0.16.0)

**Preparation**
You can test using `namespace.yml` and` policy / namespace.rego` like below.

``` yaml
# namespace.yaml
apiVersion: v1
kind: Namespace
metadata:
  name: argocd
```

``` golang
# policy/namespace.rego
package main

deny[msg] {
    input.kind = "Namespace"
    not input.metadata.name
    msg := "Namespace must have set metadata.name"
}

deny[msg] {
    input.kind = "Namespace"
    input.metadata.namespace
    msg := "Namespace must not have set metadata.namespace"
} 
```
Here's the main subject, but testing with the previous policy shows that the number of results is incorrect.

## Overview
Here's the main subject, but testing with the previous policy shows that the number of results is incorrect.

**First**, run `conftest test namespace.yaml` and see the result.

``` bash
$ conftest test namespace.yaml
--------------------------------------------------------------------------------
PASS: 1/1
WARN: 0/1
FAIL: 0/1
```

Despite defining two rules, you can see that the total number is 1.

**Next**, add a namespace to `namespace.yaml`, and then run` conftest test namespace.yaml`.
``` bash
$ cat namespace.yaml
# namespace.yaml
apiVersion: v1
kind: Namespace
metadata:
  name: argocd
  namespace: test

$ conftest test namespace.yaml
FAIL - namespace.yaml - Namespace must not have set metadata.namespace
--------------------------------------------------------------------------------
PASS: 0/1
WARN: 0/1
FAIL: 1/1
```
Still, you can see that the total number is 1.

**Finally**, delete the name from `namespace.yml` and then run` conftest test namespace.yaml`.

``` bash
$ cat namespace.yaml
# namespace.yaml
apiVersion: v1
kind: Namespace
metadata:
  # name: argocd
  namespace: test

$ conftest test namespace.yaml
FAIL - namespace.yaml - Namespace must have set metadata.name
FAIL - namespace.yaml - Namespace must not have set metadata.namespace
--------------------------------------------------------------------------------
PASS: 0/2
WARN: 0/2
FAIL: 2/2
```

In this case, the total number is 2!

Is this the expected behavior? 🤔

**Comment by jpreese** [comment]
Thanks for the descriptive report! This does appear to have been around for awhile.

**Comment by jpreese** [comment]
Resolved via #291
